### DISCUSSION TITLE
flow : payload update column

### DISCUSSION THREAD

**Discussion by nervusdm** [directus/directus#14720] [Archive]
I would like to create a flow, updating a column in the same collection.

I can't find a example or a doc about this.

Can someone tell me what should I write in the payload ?

![Capture d’écran 2022-07-28 à 14 39 58](https://user-images.githubusercontent.com/4217180/181507169-1ed10082-4094-4f8d-81f1-1728a1f4a422.png)

**Comment by nervusdm** [comment]
Oh the payload is dead simple.

```
{
    "key_column": "My Updated Operation"
}
```


But, now.. one other problem. The flow update the column, then it trigger itself... infinite loop.

**Reply by rijkvanzanten** [reply]
That shouldn't happen with "Emit Events" disabled 🤔 Can you confirm that toggle is off still?

**Reply by nervusdm** [reply]
It was off. (10000 history log in one minute lol).

But If i use Emit events "on", it's ok.

Thanks you,

**Reply by rijkvanzanten** [reply]
> But If i use Emit events "on", it's ok.

@azrikahar mind taking a look at that? That feels reversed 🔁

**Reply by azrikahar** [reply]
>  mind taking a look at that? That feels reversed 🔁

@rijkvanzanten seems like there is indeed a logical error in that when Emit Events is disabled, it is `undefined`. This is because the default value is "disabled", so the option is not set. Hence, it still satisfies `opts?.emitEvents !== false`, which is why it's still firing:

https://github.com/directus/directus/blob/6c77b8d00379c2068769c302601cf799667c1c6b/api/src/services/items.ts#L429

(Submitted the fix for this here: #14741)

That being said, I was not able to "prevent the loop" by enabling Emit Events, so I'm not sure what's up with that 🤔 since it will set `opts.emitEvents` as `true`, it _does_ still make it loop.

**Reply by rijkvanzanten** [reply]
> I was not able to "prevent the loop" by enabling Emit Events, so I'm not sure what's up with that

Maybe that was enable and then _disable_, cause then the value would be `false` instead of `undefined`
